### ISSUE TITLE
Can't mix NATS JetStream with classic NATS server

### ISSUE THREAD

**Issue by Dabada** [Redningsselskapet/nestjs-plugins#38]
**Describe the bug**
When mixing Jetstream with nats-server we got this error:
```
Error: no stream matches subject
    at JetStreamClientImpl.<anonymous> (C:\Users\mjarrai\Documents\ecomrepo\i.sale.authentication.service\node_modules\nats\nats-base-client\jsbaseclient_api.ts:97:13)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\mjarrai\Documents\ecomrepo\i.sale.authentication.service\node_modules\nats\lib\nats-base-client\jsbaseclient_api.js:19:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**To Reproduce**
In the same microservice mix nats-server with nats-jetstream

* In main.ts
```ts
 ...
  app.connectMicroservice<NatsOptions>({
    transport: Transport.NATS,
    options: {
      servers: [configService.get('nats.url')],
      name: `${configService.get('nats.name')}-subscriber`,
      queue: 'lb_queue1',
      reconnect: configService.get('nats.reconnect'),
      maxReconnectAttempts: configService.get('nats.maxReconnectAttempts'),
    },
  });

  app.connectMicroservice<CustomStrategy>({

    strategy: new NatsJetStreamServer({
      connectionOptions: {
        servers: [configService.get('nats.url')],
        name: `${configService.get('nats.name')}-durable-subscriber`,
        // name: `myservice-listener`,
        debug: true,
      },

      consumerOptions: {
        durable: 'myservice-durable', // Consumer name
        deliverTo: 'myservice-messages', // Delivery target -> push
        deliverGroup: 'myservice-group', // Delivery Queue Group
        manualAck: true,
      },

      streamConfig: {
        name: 'mystream',
        subjects: ['order.*'],
      },

    }),
  });

  await app.startAllMicroservices();
  await app.listen(servicePort, serviceIp);
```
* In controller
```ts
    ...
    @EventPattern('BULK_VALIDATION_MAIL_REQUESTED') // <- event from nats-server
    async processValidationMailRequestedBulk(@Payload() arr: { email: string, [key: string]: string }[]) {
        await this.registrationService.updateBulk(arr, State.VALIDATION_MAIL_REQUESTED);
    }

    @EventPattern('order.updated') // <- event from nats-jetstream
    public async orderUpdatedHandler(
      @Payload() data: string,
      @Ctx() context: NatsJetStreamContext,
    ) {
        console.log('received: ' + context.message.subject, data); 
        context.message.ack();
    }
```
After enabling connection debugging, I found that events from the nats server are being picked up by the jetstream event handler! In my case, the library tries to treat the "BULK_VALIDATION_MAIL_REQUESTED" event as a jetStream event.

Please check the log:
```
< PUB $JS.API.STREAM.NAMES _INBOX.69LFNR8K3IS2RS0AH4JYVL.69LFNR8K3IS2RS0AH4JZE2 44␍␊{"subject":"BULK_VALIDATION_MAIL_REQUESTED"}␍␊
> MSG _INBOX.69LFNR8K3IS2RS0AH4JYVL.69LFNR8K3IS2RS0AH4JZE2 1 106␍␊{"type":"io.nats.jetstream.api.v1.stream_names_response","total":0,"offset":0,"limit":1024,"streams":null}␍␊
Error: no stream matches subject
    at JetStreamClientImpl.<anonymous> (C:\Users\mjarrai\Documents\ecomrepo\i.sale.authentication.service\node_modules\nats\nats-base-client\jsbaseclient_api.ts:97:13)
    at Generator.next (<anonymous>)
    at fulfilled (C:\Users\mjarrai\Documents\ecomrepo\i.sale.authentication.service\node_modules\nats\lib\nats-base-client\jsbaseclient_api.js:19:58)
    at processTicksAndRejections (node:internal/process/task_queues:95:5) 
```

**Expected behavior**
Find a way to differentiate between nats-server and nats-jetstream event handlers.

**Comment by Dabada** [comment]
The problem could be in this line:
https://github.com/Redningsselskapet/nestjs-plugins/blob/38ad3f2e3b0bbd70a27b9afba61fa9412614e11a/packages/nestjs-nats-jetstream-transport/src/server.ts#L53

The filter doesn't exclude classic nats-server event handler.

**Comment by Dabada** [comment]
One solution could be to use a symbol to identify the JetStream event handler, as below:
```ts
    @EventPattern('order.updated', Symbol('JetStream'))
    public async orderUpdatedHandler(
        @Payload() data: string,
        @Ctx() context: NatsJetStreamContext,
    ) {
        console.log('received: ' + context.message.subject, data);
        context.message.ack();
    }
``` 
Then use this symbol in  server.bindEventHandlers function filter.

**Comment by Dabada** [comment]
The problem is due to an unassigned "transportId" property in the server class.

https://github.com/Redningsselskapet/nestjs-plugins/blob/38ad3f2e3b0bbd70a27b9afba61fa9412614e11a/packages/nestjs-nats-jetstream-transport/src/server.ts#L16

It should define a new transport unique identifier like this:
```ts
  readonly transportId: symbol = NATS_JETSTREAM_TRANSPORT;
```

I've forked the project for a fix.

**Comment by Dabada** [comment]
After this change, we can mix several transports like this (as documented by NestJs):

```ts
    ...
    @EventPattern('BULK_VALIDATION_MAIL_REQUESTED', Transport.NATS)
    async processValidationMailRequestedBulk(@Payload() arr: { email: string, [key: string]: string }[]) {
        await this.registrationService.updateBulk(arr, State.VALIDATION_MAIL_REQUESTED);
    }

    @EventPattern('order.updated', NATS_JETSTREAM_TRANSPORT)
    public async orderUpdatedHandler(
        @Payload() data: string,
        @Ctx() context: NatsJetStreamContext,
    ) {
        console.log(`Received: Subject=${context.message?.subject}, data=${JSON.stringify(data)} `);
        context.message.ack();
    }
```

**Comment by Dabada** [comment]
Is this project alive?

**Comment by is-blackhole** [comment]
Same issue with app having GRPC + NATS JetStream, plugin is subscribing to all event handlers, including decorated with @GrpcMethod.

**Comment by Dabada** [comment]
@is-blackhole try with this repo : https://github.com/Dabada/nestjs-plugins/tree/master/packages/nestjs-nats-jetstream-transport
It contains the above patch.

**Comment by is-blackhole** [comment]
> @is-blackhole try with this repo : https://github.com/Dabada/nestjs-plugins/tree/master/packages/nestjs-nats-jetstream-transport It contains the above patch.

Thx, I saw your PR, but how to install it as npm package?

**Comment by kenguru33** [comment]
> Is this project alive?

Sorry, but I've been a little busy lately. I just merged your pull request.

Thank you for your contribution. It is greatly appreciated.

**Comment by Dabada** [comment]
@kenguru33, you don't have to be sorry: you saved me a lot of time by making this module. So I thank you. My small contribution is nothing. 

Could you please make a new version for our friend @is-blackhole.

**Comment by kenguru33** [comment]
Just published a new npm package, version 1.3.11 :)

https://www.npmjs.com/package/@nestjs-plugins/nestjs-nats-jetstream-transport
